### ISSUE TITLE
Question Regarding GSM8k Accuracy in Qwen2.5-1.5B

### ISSUE THREAD

**Issue by yangco-le** [Yueeeeeeee/HRPO#1]
Dear authors,

First off, I'd like to express my sincere appreciation for your insightful work.

I'm currently trying to reproduce your results and have encountered an issue. When running the `qwen2.5-1.5b` model with the script provided in the `README` on the **GSM8k** dataset, I'm observing an accuracy of approximately **0.62**. This seems to be a certain difference from the results reported in your paper.

I suspect this might be related to the specific hyperparameter settings. My current execution script is as follows:

```bash
CUDA_VISIBLE_DEVICES=0 python hrpo_gsm8k.py \
  --model_name Qwen/Qwen2.5-1.5B-Instruct \
  --residual_r_min 0.98 \
  --group_size 8
```

Could you please provide some guidance on the hyperparameter configurations or suggest what might be causing this discrepancy? Any advice would be greatly appreciated\!

**Comment by feimo-49** [comment]
Hello, I'm also trying to reproduce this work!
Have you ever encountered a situation where the reward is always 0 during the whole RL training process (possibly because the responses do not contain ######)?

**Comment by yangco-le** [comment]
Hi, I haven't personally encountered a situation where the reward is consistently 0 throughout the entire RL training process . From my experience, the reward mechanism seems to be functioning as expected (qwen2.5-1.5b and qwen2.5-3b on GSM8k dataset).

**Comment by feimo-49** [comment]
Thanks for your kindly response，maybe it is due to I use llama3.1-8b-instruct model，I will try the qwen2.5-3b and carefully check if there any problems. Thanks again!



---Original---
From: "Yang ***@***.***&gt;
Date: Thu, Jun 26, 2025 20:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Yueeeeeeee/HRPO] Question Regarding GSM8k Accuracy inQwen2.5-1.5B (Issue #1)


yangco-le left a comment (Yueeeeeeee/HRPO#1)
 
Hi, I haven't personally encountered a situation where the reward is consistently 0 throughout the entire RL training process . From my experience, the reward mechanism seems to be functioning as expected (qwen2.5-1.5b and qwen2.5-3b on GSM8k dataset).

**Comment by Yueeeeeeee** [comment]
Thanks for your interest in our work, could you try different group_size and residual_r_min as used in our paper? In our log the best residual_r_min for GSM8k was 0.99 and group size for GSM8k was 4. I'm also checking further potential discrepancies between this repo and our original implementation, thanks!

**Comment by xiaoxiaostudy** [comment]
Hi, thank you for your great work!
I’m using the same execution script as mentioned in the readme:
```
CUDA_VISIBLE_DEVICES=0 python hrpo_gsm8k.py \
  --model_name Qwen/Qwen2.5-1.5B-Instruct \
  --residual_r_min 0.98 \
  --group_size 8
```
At step 1750, I'm observing an accuracy of approximately **0.66** on GSM8K.
I have a question regarding the evaluation protocol described in the paper. It mentions that the models were trained and evaluated on five datasets: GSM8K, MATH, MATH500, MMLU-STEM (MMLU-ST), and ARC-Challenge (ARC-C). I would like to confirm whether the optimal hyperparameters were tuned separately for each dataset before evaluation. Additionally, could you please share the number of training steps used for each dataset? @Yueeeeeeee

**Comment by Yueeeeeeee** [comment]
We use the last checkpoint in training to evaluate and select hyperparameters residual_r_min from 0.95, 0.98 and 0.99, we also search temperature but found this does not have a significant impact on performance. For GSM8k, we use group_size of 4 so the training steps should be about 950.
